packages/keystone/src/lib/core/queries/resolvers.ts

```diff
@@ -89,10 +87,21 @@ export async function findMany(
 ): Promise<ItemRootValue[]> {
   const orderBy = await resolveOrderBy(rawOrderBy, list, context);
 
+  // Check operation permission, throw access denied if not allowed
+  const operationAccess = await checkOperationAccess(list, context, 'query');
+  if (!operationAccess) {
+    return [];
+  }
+
   applyEarlyMaxResults(take, list);
 
+  const accessFilters = await getAccessFilters(list, context, 'query');
+  if (accessFilters === false) {
+    return [];
+  }
```

Comment:
Why do `applyEarlyMaxResults(take, list)` in the middle of these two? (and the same question for the stuff between the unique where above)

Reply:
It's pretty arbitrary, no specific motivation. I'll update it to do all the access control early updates first 👍